tests/subsys/tracing/src/main.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <ztest.h>
+
+/**
+ * @defgroup subsys_tracing_tests subsys tracing
+ * @ingroup all_tests
+ * @{
+ * @}
+ */
+
+/* size of stack area used by each thread */
+#define STACK_SIZE (512 + CONFIG_TEST_EXTRA_STACKSIZE)
+#define LOOP_TIMES 200
+
+#define dummy_test(_name) \
+	static void _name(void) \
+	{ \
+		ztest_test_skip(); \
+	}
+
+static struct k_thread thread;
+K_THREAD_STACK_DEFINE(thread_stack, STACK_SIZE);
+
+/* define semaphores */
+K_SEM_DEFINE(thread1_sem, 1, 1);
+K_SEM_DEFINE(thread2_sem, 0, 1);
+
+bool trace_thread_invoked;
+bool trace_ISR_invoked;
+bool trace_thread_idle_invoked;
+
+/* thread handle for switch */
+void thread_handle(void *p1, void *self_sem, void *other_sem)
+{
+	for (int i = 0; i < LOOP_TIMES; i++) {
+		/* take my semaphore */
+		k_sem_take(self_sem, K_FOREVER);
+
+		/* wait for a while, then let other thread have a turn. */
+		k_sleep(K_MSEC(100));
+
+		k_sem_give(other_sem);
+	}
+}
+
+/* spawn thread to generate more thread tracing data */
+void threads_loop(void)
+{
+	k_thread_create(&thread, thread_stack, STACK_SIZE,
+		thread_handle, NULL, &thread2_sem, &thread1_sem,
+		K_PRIO_PREEMPT(0), K_INHERIT_PERMS,
+		K_NO_WAIT);
+
+	thread_handle(NULL, &thread1_sem, &thread2_sem);
+}
+
+/**
+ * we need to use external tools to handle tracing data
+ * which produced by qemu/posix platform with different
+ * backends. and use popen api to execute the tools, so
```

Comment:
can we provide python scripts instead only posix code?

Reply:
Hi hakehuang,  thanks for your comment,   yes,  I thought about that,   but it's not easy to integrate with sanitycheck if I use python to implement it.

Reply:
and Do you have any suggestioons if I wan to check tracing functions with phisica board?

Reply:
I think it is similar with code implement, but yes, it requires some additional efforts. trace function is not easy way to implement, for physical board testing, we can rely on the deguger to trace, if we only trace them with uart log, then this maybe python bin-string tool.

Reply:
thanks hakehuang,  I will re-think about this.